### ISSUE TITLE
Auto exposure goes to black, then slowly fades in under certain conditions.

### ISSUE THREAD

**Issue by aFoxNamedMorris** [RobertBeckebans/RBDOOM-3-BFG#289]
I noticed that in a few places when playing through various levels and walking into a wall or door causes the screen to go near black, then slowly fade in to normal. This sometimes happens during or after teleports. Any idea what might be causing this?

**Comment by RobertBeckebans** [comment]
That is not a bug but a feature. You can disable it with r_hdrAutoExposure 0

**Comment by aFoxNamedMorris** [comment]
Oookay...

**Comment by aFoxNamedMorris** [comment]
I'm sorry, but I don't see how malfunctioning autoexposure is a "feature". The random blackouts nearly got me unfairly killed on several occasions. In a room where the camera has already adjusted to the lighting, it shouldn't go black because you tried a door or took cover.

**Comment by RobertBeckebans** [comment]
Adaptive tone mapping calculates the average luminance or so called light level of a scene and brightens or darkens the scene depending and the input. So if your scene is 2-10 times brighter than normal than the camera will lower the exposure key so you can still see things instead of getting just a white image. You can test it with the "testLight" console command. Spawn 20 extra lights and you see the effect even better. If there are flashing lights for a very short time then you can be blended like in reality.

**Comment by aFoxNamedMorris** [comment]
The problem is that the lighting in the areas in question is fairly
uniform. There was nowhere near enough contrast to trip that. Also I find
it frustrating that it takes 10 seconds or so to go from near to pitch
black back to normal in a single room with uniform lighting.

**Comment by Yetta1** [comment]
That's odd, I haven't experienced this with the auto exposure, it only darken in the far distance of darkly lit rooms and does not go to pitch black, it gives an effect more like a room at night with just some light penetrating in from say behind a curtain and underneath a door.

**Comment by aFoxNamedMorris** [comment]
Yeah. But I'm telling you: At least for me, it's malfunctioning under certain circumstances.

**Comment by DanielGibson** [comment]
Maybe you could tell us how/where to reproduce the problem?
Ideally with a video or at least a few screenshots.

**Comment by aFoxNamedMorris** [comment]
On it. I'll have them soon, hopefully.

**Comment by Yetta1** [comment]
I've tested a few maps, including Delta X and have not come across this problem. The only time it went completely black was when taking damage from a flame, which is most likely just the games effects. I tested it with a Geforce 630 on Windows 10. I'll have to install another drive to test on Linux. Might be the drivers or the build. Sometimes I get glitches, then it goes away after compiling a new executable.
